app/ui/legacy/src/main/java/com/fsck/k9/account/AccountRemoverWorker.kt

```diff
@@ -0,0 +1,45 @@
+package com.fsck.k9.account
+
+import android.content.Context
+import androidx.work.Data
+import androidx.work.OneTimeWorkRequest
+import androidx.work.OutOfQuotaPolicy
+import androidx.work.WorkManager
+import androidx.work.Worker
+import androidx.work.WorkerParameters
+
+/**
+ * A [Worker] to remove account in the background.
+ */
+class AccountRemoverWorker(
+    private val accountRemover: AccountRemover,
+    context: Context,
+    workerParams: WorkerParameters,
+) : Worker(context, workerParams) {
+
+    override fun doWork(): Result {
+        val accountUuid = inputData.getString(ARG_ACCOUNT_UUID)
+            ?: throw IllegalArgumentException("No account UUID provided")
+
+        val success = accountRemover.removeAccount(accountUuid)
+
+        return if (success) Result.success() else Result.retry()
```

Comment:
`AccountRemover.removeAccount()` returns `false` when the account UUID couldn't be found, i.e. the account has already been removed. Retrying will lead to the same result, so this operation is retried forever.

Currently there's no situation where we want to retry the account removal operation. Please change the code to always return `Result.success()`.

Reply:
Fixed in d8e09b5790147a605e4622eae6cc59a6be48e7dc